File: RadarSDK/RadarMeta.m
//
//  RadarMeta.m
//  RadarSDK
//
//  Created by Jeff Kao on 10/1/21.
//  Copyright © 2021 Radar Labs, Inc. All rights reserved.
//

#import "RadarMeta.h"

@implementation RadarMeta

+ (RadarMeta *)fromDictionary:(NSDictionary *)dict {
    RadarMeta *meta = [RadarMeta new];

    if (dict) {
        id trackingOptionsObj = dict[@"trackingOptions"];
        if (trackingOptionsObj && [trackingOptionsObj isKindOfClass:[NSDictionary class]]) {
            meta.trackingOptions = [RadarTrackingOptions trackingOptionsFromDictionary:trackingOptionsObj];
        }
        id sdkConfigurationObj = dict[@"sdkConfiguration"];
        if (sdkConfigurationObj && [sdkConfigurationObj isKindOfClass:[NSDictionary class]]) {
            meta.sdkConfiguration = [[RadarSdkConfiguration alloc] initWithDict:sdkConfigurationObj];
        }
    }

    return meta;
}

@end

File: RadarSDK/RadarMeta.swift
//
//  RadarMeta.swift
//  RadarSDK
//
//  Copyright © 2026 Radar Labs, Inc. All rights reserved.
//

import Foundation

@objc(RadarMeta) @objcMembers
class RadarMeta: NSObject {
    var trackingOptions: RadarTrackingOptions?
    var sdkConfiguration: RadarSdkConfiguration?

    static func from(dictionary dict: [String: Any]?) -> RadarMeta? {
        let meta = RadarMeta()

        if let dict {
            if let trackingOptionsDict = dict["trackingOptions"] as? [String: Any] {
                meta.trackingOptions = RadarTrackingOptions(from: trackingOptionsDict)
            }
            if let sdkConfigurationDict = dict["sdkConfiguration"] as? [String: Any] {
                meta.sdkConfiguration = RadarSdkConfiguration(dict: sdkConfigurationDict)
            }
        }

        return meta
    }
}
